diff --git a/src/lib.rs b/src/lib.rs
index 2fe30b9..88cd566 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1121,6 +1121,13 @@ impl Build {
                     .ok_or_else(|| Error::new(ErrorKind::InvalidArgument, "parent() failure"))?
                     .to_string_lossy();
                 let mut hasher = hash_map::DefaultHasher::new();
+                let out_dir = self.get_out_dir().expect("Could not get out dir");
+
+                let prefix = out_dir.parent().expect("Could not get parent");
+                let prefix: &str = &prefix.to_string_lossy();
+
+                let err = format!("could not strip prefix {prefix} from {dirname}");
+                let dirname = dirname.strip_prefix(prefix).expect(&err);
                 hasher.write(dirname.to_string().as_bytes());
                 dst.join(format!("{:016x}-{}", hasher.finish(), basename))
                     .with_extension("o")
