# Patch for issue: https://github.com/bytecodealliance/wasmtime/pull/10334
diff --git a/cranelift/assembler-x64/src/lib.rs b/cranelift/assembler-x64/src/lib.rs
index 7e98e0226..b942d0d4a 100644
--- a/cranelift/assembler-x64/src/lib.rs
+++ b/cranelift/assembler-x64/src/lib.rs
@@ -77,8 +77,8 @@ pub use mem::{Amode, DeferredTarget, GprMem, Scale};
 pub use reg::{Gpr, NonRspGpr, Size};
 pub use rex::RexFlags;
 
-/// List the files generated to create this assembler.
-#[must_use]
-pub fn generated_files() -> Vec<std::path::PathBuf> {
-    include!(concat!(env!("OUT_DIR"), "/generated-files.rs"))
+/// Return the contents of `assembler-definitions.isle` which is needed to
+/// generate the x64 instruction selector.
+pub fn assembler_definitions_isle_contents() -> Vec<u8> {
+    include_bytes!(concat!(env!("OUT_DIR"), "/assembler-definitions.isle")).to_vec()
 }
