# Fix for https://github.com/bytecodealliance/wasmtime/issues/9553
diff --git a/cranelift/srcgen/src/lib.rs b/cranelift/srcgen/src/lib.rs
index 019fba9b8..a36184a30 100644
--- a/cranelift/srcgen/src/lib.rs
+++ b/cranelift/srcgen/src/lib.rs
@@ -238,7 +238,6 @@ impl Formatter {
         directory: &std::path::Path,
     ) -> Result<(), error::Error> {
         let path = directory.join(&filename);
-        eprintln!("Writing generated file: {}", path.display());
         let mut f = fs::File::create(path)?;
 
         for l in self.lines.iter().map(|l| l.as_bytes()) {
