diff --git a/cranelift/isle/isle/build.rs b/cranelift/isle/isle/build.rs
index 2b71758a8..c14d8c5b0 100644
--- a/cranelift/isle/isle/build.rs
+++ b/cranelift/isle/isle/build.rs
@@ -20,8 +20,14 @@ fn main() {
 }
 
 fn emit_tests(out: &mut String, dir_name: &str, runner_func: &str) {
+    let mut test_files = vec![];
     for test_file in std::fs::read_dir(dir_name).unwrap() {
-        let test_file = test_file.unwrap().file_name().into_string().unwrap();
+        test_files.push(test_file.unwrap().file_name().into_string().unwrap());
+    }
+
+    test_files.sort();
+
+    for test_file in test_files {
         if !test_file.ends_with(".isle") {
             continue;
         }
