# workaround for https://github.com/tweag/rules_haskell/issues/2254
diff --git a/haskell/private/workspace_utils.bzl b/haskell/private/workspace_utils.bzl
index df851329..980b3e16 100644
--- a/haskell/private/workspace_utils.bzl
+++ b/haskell/private/workspace_utils.bzl
@@ -19,6 +19,8 @@ def execute_or_fail_loudly(
         environment = environment,
         quiet = True,
         working_directory = working_directory,
+        # triple the default timeout because e.g. copying GHC is slow (1.5G)
+        timeout = 3 * 600,
     )
     if exec_result.return_code != 0:
         arguments = [_as_string(x) for x in arguments]
